hathor/p2p/sync_v2/transaction_streaming_client.py

```diff
@@ -100,6 +100,9 @@ def resume(self) -> Deferred[StreamEnd]:
 
     def fails(self, reason: 'StreamingError') -> None:
         """Fail the execution by resolving the deferred with an error."""
+        if self._deferred.called:
+            self.log.warn('already failed before', new_reason=str(reason))
```

Comment:
Maybe `repr(reason)`? I feel it's worthy logging the name of the exception.

Reply:
Done.

Reply:
Thanks!